service/storage_service.cc

```diff
@@ -1279,6 +1283,11 @@ class join_node_rpc_handshaker : public service::group0_handshaker {
     future<> pre_server_start(const group0_info& g0_info) override {
         rtlogger.info("join: sending the join request to {}", g0_info.ip_addr);
 
+        co_await utils::get_local_injector().inject("crash_before_group0_join", [](auto& handler) -> future<> {
+            // This wait ensures that node gossips its state before crashing.
+            co_await handler.wait_for_message(db::timeout_clock::now() + std::chrono::minutes(5));
+            throw std::runtime_error("deliberately crashed for orphan remover test");
+        });
```

Comment:
I'm not even sure my idea works and it seems like too much effort for too little reward on a second thought. Let's not overcomplicate it, especially since this PR already has 100+ comments.